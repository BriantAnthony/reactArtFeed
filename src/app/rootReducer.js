import { combineReducers } from 'redux';
import userReducer from '../user/userReducer';
import cardReducer from '../cardFeed/cardReducer';

const rootReducer = combineReducers({
  app: userReducer,
  artCard: cardReducer
})

export default rootReducer;