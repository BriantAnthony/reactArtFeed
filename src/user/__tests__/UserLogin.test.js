import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../../app/rootReducer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {
  Card, 
  CardActions, 
  CardHeader, 
  CardText
} from 'material-ui/Card';
import Login from '../UserLogin';
import renderer from 'react-test-renderer';

const store = createStore(rootReducer);
const form = {
  email: 'briantacampbell@gmail.com',
  password: 'demo'
};

it('renders without crashing', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <Provider store={store}>
        <Login/>
      </Provider>
    </MuiThemeProvider>
  ).toJSON();
  //console.log('tree: ', tree);
  expect(tree).toMatchSnapshot();
});
