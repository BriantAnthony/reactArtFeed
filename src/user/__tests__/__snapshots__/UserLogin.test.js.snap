exports[`test renders without crashing 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "marginLeft": "auto",
      "marginRight": "auto",
      "muiPrepared": true,
      "padding": 10,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": 400,
      "zIndex": 1,
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }>
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }>
          Login
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }>
          Demo Account
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }>
      <div>
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }>
          <label
            className={undefined}
            htmlFor="undefined-email-Email-30749"
            onTouchTap={undefined}
            style={
              Object {
                "color": "#ff9800",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(0.75) translate(0, -28px)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }>
            Email
          </label>
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            email
          </div>
          <input
            defaultValue="briantacampbell@gmail.com"
            disabled={false}
            id="undefined-email-Email-30749"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text" />
          <div>
            <hr
              style={
                Object {
                  "borderBottom": "solid 1px",
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              } />
            <hr
              style={
                Object {
                  "borderBottom": "solid 2px",
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              } />
          </div>
        </div>
        <br />
        <div
          className={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }>
          <label
            className={undefined}
            htmlFor="undefined-Password-Password-54626"
            onTouchTap={undefined}
            style={
              Object {
                "color": "#ff9800",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(0.75) translate(0, -28px)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }>
            Password
          </label>
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            Password
          </div>
          <input
            defaultValue="demo"
            disabled={false}
            id="undefined-Password-Password-54626"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="password" />
          <div>
            <hr
              style={
                Object {
                  "borderBottom": "solid 1px",
                  "borderColor": "#e0e0e0",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              } />
            <hr
              style={
                Object {
                  "borderBottom": "solid 2px",
                  "borderColor": "#00bcd4",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              } />
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }>
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": "2px",
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                   0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        <button
          disabled={false}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button">
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}>
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }>
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }>
                Login
              </span>
            </div>
          </div>
        </button>
      </div>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Forgot Password
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
